src/blockstorage/sequential_files.cpp

```diff
@@ -154,7 +154,7 @@ uint64_t CalculateCurrentUsage()
 /* Prune a block file (modify associated database entries)*/
 void PruneOneBlockFile(const int fileNumber)
 {
-    READLOCK(cs_mapBlockIndex);
+    RECURSIVEREADLOCK(cs_mapBlockIndex);
     for (BlockMap::iterator it = mapBlockIndex.begin(); it != mapBlockIndex.end(); ++it)
```

Comment:
What deadlock are you trying to fix?

Reply:
it is just this lock. it deadlocks itself. this started happening when we switched from a recursive mutex to a shared mutex. some of the blockdb sync code would recurively lock itself which now gets stuck.

I want to test an IBD of dev prior to this PR and then with this PR to see how the rsm performs. is there a huge performance hit or not?